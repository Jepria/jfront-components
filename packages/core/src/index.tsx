export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
