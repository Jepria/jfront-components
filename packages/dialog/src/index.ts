export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'