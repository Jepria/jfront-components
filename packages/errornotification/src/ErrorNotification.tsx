import React from "react"
import { ErrorDialog } from "@jfront/ui-dialog"
import {
  ACCESS_DENIED,
  AUTHORIZATION_FAILED,
  BadRequest,
  BAD_REQUEST,
  NetworkError,
  NotFound,
  NOT_FOUND,
  ServerError,
} from "@jfront/core-rest"

export interface ErrorNotificationProps {
  error?: NetworkError | Error | string | undefined
  header?: string
  log?: (error: any, info?: any) => void
  render?: (error: NetworkError | Error | string | undefined) => React.ReactNode
  children?: React.ReactNode
}

interface ErrorNotificationState {
  error: NetworkError | Error | string | undefined
}

export class ErrorNotification extends React.Component<
  ErrorNotificationProps,
  ErrorNotificationState
> {
  constructor(props: ErrorNotificationProps) {
    super(props)
    this.state = { error: props.error }
    window.onerror = (msg, url, line, col, error) => {
      // Note that col & error are new to the HTML 5 spec and may not be
      // supported in every browser.
      let extra = !col ? "" : "\ncolumn: " + col
      extra += !error ? "" : "\nerror: " + error
      const log = "Error: " + msg + "\nurl: " + url + "\nline: " + line + extra
      // You can view the information in an alert to see things working like this:
      console.error(log)
      if (this.props.log) {
        if (error) {
          this.props.log(error, log)
        } else {
          this.props.log(log)
        }
      }
      const suppressErrorAlert = true
      // If you return true, then error alerts (like in older versions of
      // Internet Explorer) will be suppressed.
      if (error) {
        this.setState({ error })
      } else {
        this.setState({ error: log })
      }
      return suppressErrorAlert
    }

    window.onunhandledrejection = (e: PromiseRejectionEvent) => {
      console.error(e)
      if (this.props.log) {
        this.props.log(e)
      }
      this.setState({ error: e.reason })
    }
  }

  static getDerivedStateFromError(error: any) {
    return { error }
  }

  componentDidCatch(error: any, info: any) {
    console.error(error, info)
    if (this.props.log) {
      this.props.log(error, info)
    }
  }

  render() {
    const { header = "Ошибка", children } = this.props
    const { error } = this.state
    if (error) {
      if (this.props.render) {
        return this.props.render(error)
      }
      let errorId, errorCode, errorMessage, errorDescription
      if (typeof error === "string") {
        errorMessage = error
      } else if ((error as NetworkError).type) {
        switch ((error as NetworkError).type) {
          case BAD_REQUEST: {
            errorCode = BAD_REQUEST
            errorDescription = "Неверный формат HTTP запроса"
            errorMessage = JSON.stringify(
              (error as BadRequest).constraintViolations,
            )
            break
          }
          case AUTHORIZATION_FAILED: {
            errorCode = AUTHORIZATION_FAILED
            errorDescription = "Ошибка авторизации"
            break
          }
          case ACCESS_DENIED: {
            errorCode = ACCESS_DENIED
            errorDescription = "Недостаточно прав доступа"
            break
          }
          case NOT_FOUND: {
            errorCode = NOT_FOUND
            errorDescription = (error as NotFound).url + " URL not found"
            break
          }
          default: {
            errorId = (error as ServerError).errorId
            errorCode = (error as ServerError).errorCode
            errorMessage = (error as ServerError).errorMessage
            break
          }
        }
      } else {
        errorMessage = (error as Error).stack
        errorDescription = (error as Error).message
      }
      return (
        <>
          <ErrorDialog
            header={header}
            visible={error !== undefined}
            errorId={errorId}
            errorCode={errorCode}
            errorMessage={errorMessage}
            errorDescription={errorDescription}
            onClose={() => this.setState({ error: undefined })}
          />
          {children}
        </>
      )
    } else {
      return children
    }
  }
}
