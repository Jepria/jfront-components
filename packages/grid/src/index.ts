export * from './common'
export * from './jepgrid'
