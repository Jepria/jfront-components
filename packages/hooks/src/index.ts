export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
