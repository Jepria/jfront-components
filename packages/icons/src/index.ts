export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
