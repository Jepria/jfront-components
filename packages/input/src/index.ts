export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
