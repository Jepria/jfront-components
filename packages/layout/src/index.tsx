import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
