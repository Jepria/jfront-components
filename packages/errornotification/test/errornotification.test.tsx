import React from "react"
import { render } from "react-dom"
import { ErrorNotification } from "../src"
import { fireEvent, screen } from "@testing-library/react"

const container = document.createElement("div")
document.body.appendChild(container)

test.skip("Matches snapshot ", () => {
  render(
    <ErrorNotification error={new Error("Test error")}>UI</ErrorNotification>,
    container,
  )
  expect(document.body).toMatchSnapshot()
})

test("Error capture", () => {
  const ErrorTest = () => {
    throw new Error("test error")
  }
  render(
    <ErrorNotification>
      <ErrorTest />
    </ErrorNotification>,
    container,
  )
  expect(screen.queryAllByText("test error")).toHaveLength(1)
})
