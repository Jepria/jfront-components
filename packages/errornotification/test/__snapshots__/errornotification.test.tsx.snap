// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<body>
  <div>
    <div
      class="sc-AxhCb fJYEYh"
    />
    <div
      class="sc-Axmtr eBZWUw"
    >
      <div
        class="sc-AxhUy dybvAb sc-fznyAO SlMew"
      >
        <header
          class="sc-AxiKw kwuhTE"
        >
          Ошибка
        </header>
        <section
          class="sc-AxgMl hENNCS"
          style="flex-direction: row;"
        >
          <div
            class="sc-fzqNJr chjqeJ"
          >
            <span>
              <svg
                class="sc-fzokOt jZsFTK"
                enable-background="new 0 0 455.111 455.111;"
                spacing="preserve"
                version="1.1"
                viewBox="0 0 455.111 455.111"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <circle
                  cx="227.556"
                  cy="227.556"
                  fill="#E24C4B"
                  r="227.556"
                />
                <path
                  d="M455.111,227.556c0,125.156-102.4,227.556-227.556,227.556c-72.533,0-136.533-32.711-177.778-85.333c38.4,31.289,88.178,49.778,142.222,49.778c125.156,0,227.556-102.4,227.556-227.556c0-54.044-18.489-103.822-49.778-142.222C422.4,91.022,455.111,155.022,455.111,227.556z"
                  fill="#D1403F"
                />
                <path
                  d="M331.378,331.378c-8.533,8.533-22.756,8.533-31.289,0l-72.533-72.533l-72.533,72.533c-8.533,8.533-22.756,8.533-31.289,0c-8.533-8.533-8.533-22.756,0-31.289l72.533-72.533l-72.533-72.533c-8.533-8.533-8.533-22.756,0-31.289c8.533-8.533,22.756-8.533,31.289,0l72.533,72.533l72.533-72.533c8.533-8.533,22.756-8.533,31.289,0c8.533,8.533,8.533,22.756,0,31.289l-72.533,72.533l72.533,72.533C339.911,308.622,339.911,322.844,331.378,331.378z"
                  fill="#FFFFFF"
                />
              </svg>
            </span>
          </div>
          <div
            class="sc-fzoyAV biuXzx"
          >
            <div>
              Test error
            </div>
          </div>
        </section>
        <footer
          class="sc-AxheI eZFndM"
        >
          <button
            class="sc-fzoLsD dwsGTu sc-fzpans vXdVQ"
            id="id1"
            type="button"
            value="OK"
          >
            OK
          </button>
          <button
            class="sc-fzoLsD dwsGTu sc-fzpans vXdVQ"
            id="id2"
            type="button"
            value="Показать детали"
          >
            Показать детали
          </button>
        </footer>
      </div>
    </div>
    UI
  </div>
</body>
`;
